React/Executors/RCTJSCExecutor.m

```diff
@@ -324,6 +324,20 @@ - (void)setUp
                                                  name:event
                                                object:nil];
   }
+
+  // Inject handler used by HMR
+  [self addSynchronousHookWithName:@"nativeInjectHMRUpdate" usingBlock:^(NSString *sourceCode, NSString *sourceCodeURL) {
+    RCTJSCExecutor *strongSelf = weakSelf;
+    if (!strongSelf.valid) {
+      return;
+    }
+
+    JSStringRef execJSString = JSStringCreateWithUTF8CString(sourceCode.UTF8String);
+    JSStringRef jsURL = JSStringCreateWithUTF8CString(sourceCodeURL.UTF8String);
+    JSEvaluateScript(strongSelf->_context.ctx, execJSString, NULL, jsURL, 0, NULL);
```

Comment:
Yeah, when I initially implemented this the code was originally on `executeApplicationScript` because it looked like the context was cleared after running this method. Maybe the problem was that we were assigning the sync function directly to the context. This is the right place and the right way to define the callback :)